### DISCUSSION TITLE
i have zero clue, please help.

### DISCUSSION THREAD

**Discussion by veyg208** [LibreHardwareMonitor/LibreHardwareMonitor#1543] [Q&A]
just wanting to use the pull request to edit lhw so that i can get my motherboard to work in FanControl... X870E-E for reference. Theres no guide i could find on how to get this done.

**Comment by d-e-n-n-i-s-t** [comment]
I agree, I recently created a pull request for my motherboard and it was pretty hard to do with little guidance. A little guide somewhere would definitely help.

First, I'd recommend you try to find the name of the SuperIO chip that's used on your motherboard. I found mine in HWiNFO64 and it was also written in LHM. Most (if not all?) SuperIO chips are supported already, you just have to adjust the setup because your motherboard manufacturer has its own layout of fan headers, voltages, temps etc

You need to sort out which indices correlate to which voltage/temp/fan (**for example** Vcore is voltage nr 5, SoC is voltage nr 4 etc). Voltages can also accept the parameters Ri, Rf, and Vf, and you may need to adjust them to get a correct voltage readout. I also recommend taking a quick look at Control, Fan, Temperature and Voltage files at "LibreHardwareMonitorLib/Hardware/Motherboard"

Maybe you can follow the path I took in #1532 .

**Comment by SalvatoreSatori** [comment]
sitting in the same boat, somebody updated the the motherboard info for all x870 MB like mine (x870-i) they way I understood this I somehow need to download the repository (which I did) and then compile the entire thing in order to get the new DLL LIB and XML file, these then get copied into fancontrol and it should run ?!? at least that's what I think lol, so far though I could not find any way to compile the "repository" so I am totally lost...Could please some of the GITHUB Gods help us NEWBS? Tx oh mighty ones :)

**Reply by d-e-n-n-i-s-t** [reply]
Look at the reply I made above yours, and follow the link. You basically fork the repo, apply the changes you want and then change the "master.yml" file so Github actually builds your fork for you. Then, after you confirmed it works, you revert "master.yml" and create a pull request from your fork.

You can look up how to exactly do this in the pull request I linked in my comment above.
